mitodl/open-discussions
# Cc/yarn workspaces

#### Pre-Flight checklist


#### What are the relevant tickets?
Related to https://github.com/mitodl/hq/discussions/95

#### What's this PR do?
This PR reorganizes the frontend code in open-discussions to use yarn workspaces, similar to https://github.com/mitodl/mitxonline/pull/514.

There are no changes to application code. The only changes in this PR are to the build process and javascript dependency versions.

#### How should this be manually tested?
Restart docker and:
1. Visit `http://yourname.od.odl.local:8063/`. You should see open-discussions with no changes. Changing the js source code in `frontends/open-discussions/src` should live reload the site.
2. In your env file, set `DEV_WORKSPACE='infinite-corridor'`
3. recreate the docker containers: `docker compose up -d`
4. Visit `http://yourname.od.odl.local:8063/` again. You should see a placeholder site for infinite corridor. Make changes in the source code (`frontends/infinite-corridor/src`) if you want and observe the nice live reload.

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)
